lib/tapioca/dsl/compilers/frozen_record.rb

```diff
@@ -72,13 +72,17 @@ def decorate
           attributes = constant.attributes
           return if attributes.empty?
 
+          instance = constant.first
+
           root.create_path(constant) do |record|
             module_name = "FrozenRecordAttributeMethods"
 
             record.create_module(module_name) do |mod|
               attributes.each do |attribute|
+                return_type = instance.attributes[attribute].class.name
```

Comment:
This approach isn't sufficient to be a complete type for all the collection types YAML supports (e.g. `Array` and `Hash`) since we aren't (and shouldn't) recursively check every member of the collection to create a union of all possible types.

I think this is "fine" since [Sorbet itself](https://sorbet.org/docs/stdlib-generics) removed this functionality during runtime due to performance problems.

Having these be `T::Array[T.untyped]` and `T::Hash[T.untyped, T.untyped]` seems "OK" and devs can further refine the type with `T.cast` at runtime.